operations/configuration_management/NCN_Node_Personalization.md

```diff
@@ -29,62 +66,151 @@ The following procedure describes how to correctly edit the `sat bootprep` files
     - name: ncn-personalization
     ```
 
-1. In the case of an upgrade, execute the following two substeps:
+### 2. Edit CPE and Analytics layers
+
+This section is required if this procedure is being performed as part of a CSM upgrade.
+If that is not the case, then skip ahead to [Update CFS configuration and components](#4-update-cfs-configuration-and-components).
+
+In this section, the `management-bootprep-node-personalization.yaml` file is modified to specify the versions of CPE and Analytics that
+are currently in use on the system. This is necessary because the new versions of CPE and Analytics have not yet been installed at this
+time in the CSM upgrade procedure.
+
+1. (`ncn-m#`) Acquire a copy of the current CPE and Analytics products CFS configuration values already in use.
+
+    There are multiple ways to do this. The following is an example of finding these values by looking at the CFS configuration of a worker NCN on the system.
```

Comment:
Do you have any concern here about the potential for some NCNs to have different configurations?

Not saying it has to be done with SAT and as you say it can be done different ways, but using `sat status` here would tell you the desired configurations for all the NCNs at once.

```
ncn-m001:~ # sat status --filter role=management --fields xname,desiredconfig
+---------------+---------------------+
| xname         | Desired Config      |
+---------------+---------------------+
| x3000c0s1b0n0 | ncn-personalization |
| x3000c0s2b0n0 | ncn-personalization |
| x3000c0s3b0n0 | ncn-personalization |
| x3000c0s4b0n0 | ncn-personalization |
| x3000c0s5b0n0 | ncn-personalization |
| x3000c0s6b0n0 | ncn-personalization |
| x3000c0s7b0n0 | ncn-personalization |
| x3000c0s8b0n0 | ncn-personalization |
| x3000c0s9b0n0 | ncn-personalization |
+---------------+---------------------+
```

And if you wanted to be really fancy:
```
ncn-m001:~ # sat status --filter role=management --fields desiredconfig --format json | jq -r '.[]."Desired Config"' | uniq
ncn-personalization
```

Scrolling down in the procedure a little I see that it doesn't REALLY matter that much because the procedure will have you update the `desiredConfig` for all the nodes at once to this config anyway, so by the end they will all have the same config.

Reply:
Using sat for this definitely seems simpler, so I'll make that edit. And if they are using multiple configurations, then they'd probably want to know that so they could make sure that they are using the one they wanted as the source. Good call.